### ISSUE TITLE
SelectOneMenu: accessibility violation

### ISSUE THREAD

**Issue by GabrieleFrau** [primefaces/primefaces#6042]
**Describe the defect**
`p:selectOneMenu` doesn't follow WAI-ARIA Specs

**Environment:**
 - PF Version: _6.2.27_
 - JSF + version: _Mojarra N.D._
 - Affected browsers: _ALL_

**To Reproduce**
1. Load xhtml page
2. Primefaces generates incorrect html element

**Expected behavior**

Every Input with id myid_focus shouldn't have an `aria-expanded` attribute

> When ARIA attributes are used on HTML elements that are not in accordance with WAI-ARIA 1.1, they conflict with the semantics of the elements which can cause assistive technology products report nonsensical user interface (UI) information that does not represent the actual UI of the document.

https://dequeuniversity.com/rules/axe/3.5/aria-allowed-attr?application=AxeChrome

**Example XHTML**
```html
<h:form id="frmTest">
    <p:selectOneMenu id="myid" value="#{bean.variable}">
        <f:selectItems value="#{bean.variables}" />
    </p:selectOneMenu>
</h:form>
```
generates
```html
<form id="frmTest" name="frmTest" method="post" action="/tributi/utente/welcome.xhtml" enctype="application/x-www-form-urlencoded">
	<input type="hidden" name="frmTest" value="frmTest">
	<div id="frmTest:myid" class="ui-selectonemenu ui-widget ui-state-default ui-corner-all ui-state-hover" role="combobox" aria-haspopup="true" aria-expanded="true" aria-owns="frmTest:myid_items" style="min-width: 24px;">
		<div class="ui-helper-hidden-accessible">
			<input id="frmTest:myid_focus" name="frmTest:myid_focus" type="text" autocomplete="off" aria-expanded="true" aria-autocomplete="list" aria-disabled="false">
		</div>
		<div class="ui-helper-hidden-accessible">
			<select id="frmTest:myid_input" name="frmTest:myid_input" tabindex="-1" aria-hidden="true"/>
		</div>
		<label id="frmTest:myid_label" class="ui-selectonemenu-label ui-inputfield ui-corner-all">&nbsp;</label>
		<div class="ui-selectonemenu-trigger ui-state-default ui-corner-right ui-state-hover">
			<span class="ui-icon ui-icon-triangle-1-s ui-c"/>
		</div>
	</div>
	<input type="hidden" name="javax.faces.ViewState" id="j_id1:javax.faces.ViewState:1" value="-856014644197753318:920539168267910855" autocomplete="off">
</form>
```
**Current workaround**
```html
<h:form id="frmTest">
    <p:selectOneMenu id="myid" value="#{bean.variable}">
        <f:selectItems value="#{bean.variables}" />
    </p:selectOneMenu>
    <script>
        document.getElementById("frmtest:myid_focus").removeAttribute("aria-expanded");
    </script>
</h:form>
```

**Comment by tandraschko** [comment]
Did you try the master if its already fixed?

Please read our issue template:
**NOTE:** PrimeFaces test can easily be set to use the latest `master-SNAPSHOT` of what is in GitHub. Please test with that before reporting a bug that may already be fixed.

**Comment by melloware** [comment]
@GabrieleFrau As Thomas said we have done a lot of ARIA work in PF7 and 8 so there is a good chance this is already fixed.  Can you test in the Showcase: https://www.primefaces.org/showcase/ui/input/oneMenu.xhtml and let us know if anything is still wrong?

**Comment by melloware** [comment]
You are right it still does have aria-expanded and it should not.

**Comment by melloware** [comment]
To fix this correctly right now you can add this JS to your application.
```javascript
PrimeFaces.widget.SelectOneMenu.prototype._show = function() {
    var $this = this;
    this.alignPanel();

    this.panel.css('z-index', ++PrimeFaces.zindex);

    if ($.browser.msie && /^[6,7]\.[0-9]+/.test($.browser.version)) {
        this.panel.parent().css('z-index', PrimeFaces.zindex - 1);
    }

    if (this.cfg.effect !== 'none') {
        this.panel.show(this.cfg.effect, {}, this.cfg.effectSpeed, function() {
            PrimeFaces.scrollInView($this.itemsWrapper, $this.getActiveItem());

            if ($this.cfg.filter)
                $this.focusFilter();
        });
    } else {
        this.panel.show();
        PrimeFaces.scrollInView(this.itemsWrapper, this.getActiveItem());

        if ($this.cfg.filter)
            this.focusFilter(10);
    }

    //value before panel is shown
    this.preShowValue = this.options.filter(':selected');
    this.jq.attr('aria-expanded', true);
};
```

**Comment by GabrieleFrau** [comment]
> Did you try the master if its already fixed?
> 
> Please read our issue template:
> **NOTE:** PrimeFaces test can easily be set to use the latest `master-SNAPSHOT` of what is in GitHub. Please test with that before reporting a bug that may already be fixed.

I was under the impression that Primefaces 6 was still supported and as such should have this kind of fixes backported. Am I wrong in that assumption? Should I start the QA process with Primefaces 8 instead?

**Comment by tandraschko** [comment]
it's a bit complicated :D (I'm not a member of PrimeTek!)

This here is the community issue tracker. Bugs are fixed here in the github master and partly backported to the elite releases from PrimeTek, which are maintained in a private PrimeTek repository.
Which issues are packported is the decision by PrimeTek.
PRO users can directly influence issues to be packported in elite releases.
The current supported versions are 7.0.x and 8.0.x.

So it's actually a required information for us, the community, to check if its already fixed in trunk.

And yes, please try to upgrade :) It contains sooo many improvements.

**Comment by kukel** [comment]
sooo is an understatement... soooooooo is better ;-)
